src/ocean_emulators/models/modules/encoder.py

```diff
@@ -0,0 +1,90 @@
+# Sources inspired by the following implementations:
+# - https://github.com/microsoft/aurora/blob/main/aurora/model/patchembed.py
+# - https://github.com/lucidrains/vit-pytorch
+from typing import Any
+
+import torch
+from einops import rearrange
+from jaxtyping import Float
+from perceiver_pytorch import Perceiver
+from torch import nn
+
+from ocean_emulators.constants import Input
+
+
+class PerceiverEncoder(nn.Module):
+    """A perceiver-based encoder for Samudra's flattened data (a whole column of the ocean, with history).
+
+    Args:
+        n_channels (int): the number of input channels (roughly:  time x variable x (surface + depths)).
+        patch_size (int): the size of the patches to embed. Patches must evenly divide the input grid.
+        embed_dim (int): size of the latent dimension.
+        perceiver_depth (int): depth of the perceiver module core.
+        perceiver_kwargs (dict): keyword arguments to pass to the perceiver module. The docs for this module
+          can be found here: https://github.com/lucidrains/perceiver-pytorch#usage.
+    """
+
+    def __init__(
+        self,
+        n_channels: int,
+        patch_size: int | tuple[int, int],
+        embed_dim: int,
+        perceiver_depth: int,
+        **perceiver_kwargs: dict[str, Any],
```

Comment:
```suggestion
        **perceiver_kwargs: Any,
```
The annotation here is on the values, not on the type of the resulting dict. Alternatively, do we need to expose this? Some of these shouldn't be passed (eg the ones we're passing ourselves).

Reply:
Yeah, that's a fair point -- in some sense these are implementation details. I thought we might want to have this be config-driven at first, but it probably is best to manage this via code.